PhysX Gem can't be used as build dependency in engine SDK

**Describe the bug**
If the PhysX Gem is added as a build dependency in CMakeLists.txt, generating the project files with CMake fail with the following error:
```
CMake Error at D:/dev/open3d/o3de-install/cmake/LYWrappers.cmake:174 (add_library):
  Target "PhysXTest" links to target
  "$<TARGET_PROPERTY:Gem::LmbrCentral,INTERFACE_LINK_LIBRARIES>" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
Call Stack (most recent call first):
  Code/CMakeLists.txt:42 (ly_add_target)

CMake Error at D:/dev/open3d/o3de-install/cmake/LYWrappers.cmake:174 (add_library):
  Target "PhysXTest" links to target
  "$<TARGET_PROPERTY:Gem::LmbrCentral,INTERFACE_LINK_LIBRARIES>" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
Call Stack (most recent call first):
  Code/CMakeLists.txt:42 (ly_add_target)

CMake Error at D:/dev/open3d/o3de-install/cmake/LYWrappers.cmake:174 (add_library):
  Target "PhysXTest" links to target
  "$<TARGET_PROPERTY:Gem::LmbrCentral,INTERFACE_LINK_LIBRARIES>" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
Call Stack (most recent call first):
  Code/CMakeLists.txt:42 (ly_add_target)

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Build the engine as an SDK
2. Create a new project from the Project Manager
3. In the projects CMakeLists.txt, add `Gem::PhysX` to BUILD_DEPENDENCIES:
```
ly_add_target(
    NAME PhysXTest.Static STATIC
    NAMESPACE Gem
    FILES_CMAKE
        physxtest_files.cmake
        ${pal_dir}/physxtest_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
    INCLUDE_DIRECTORIES
        PUBLIC
            Include
    BUILD_DEPENDENCIES
        PRIVATE
            AZ::AzGameFramework
            Gem::Atom_AtomBridge.Static
            Gem::PhysX
)
```
(Doesn't matter if it's in `<project>` or `<project>.Static` Target

**Expected behavior**
Successful cmake configure and build

**Additional context**
A workaround is commenting out a line in \o3de-install\Gems\PhysX\Code\CMakeLists.txt (4th last line):
```
ly_add_target(
    NAME PhysX IMPORTED GEM_MODULE
    NAMESPACE Gem
    COMPILE_DEFINITIONS
        INTERFACE
PHYSX_ENABLE_MULTI_THREADING
    INCLUDE_DIRECTORIES
        INTERFACE
${LY_ROOT_FOLDER}/include/Gems/PhysX/Code/Include

    BUILD_DEPENDENCIES
        INTERFACE
Gem::PhysX.Static
#$<TARGET_PROPERTY:Gem::LmbrCentral,INTERFACE_LINK_LIBRARIES>
    RUNTIME_DEPENDENCIES
Gem::LmbrCentral
)
```

There are other references to `Gem::LmbrCentral,INTERFACE_LINK_LIBRARIES`, but it looks like this line is the only one that causes the error.

SDK build of the development branch, commit ec02696.

--- a/cmake/LYWrappers.cmake
+++ b/cmake/LYWrappers.cmake
@@ -380,12 +380,14 @@ function(ly_delayed_target_link_libraries)
             cmake_parse_arguments(ly_delayed_target_link_libraries "" "" "${visibilities}" ${delayed_link})
 
             foreach(visibility ${visibilities})
-                foreach(item ${ly_delayed_target_link_libraries_${visibility}})
+                foreach(alias_item ${ly_delayed_target_link_libraries_${visibility}})
 
-                    if(TARGET ${item})
-                        get_target_property(item_type ${item} TYPE)
+                    if(TARGET ${alias_item})
+                        get_target_property(item_type ${alias_item} TYPE)
+                        ly_de_alias_target(${alias_item} item)
                     else()
                         unset(item_type)
+                        set(item ${alias_item})
                     endif()
 
                     if(item_type STREQUAL MODULE_LIBRARY)

